### ISSUE TITLE
Hugo 0.55.* has broken Shortcode.Inner for <table> elements

### ISSUE THREAD

**Issue by yawpitch** [gohugoio/hugo#5902]
Updating to Hugh 0.55 (any version, including 0.56.0-DEV) has broken shortcodes that wrap a markdown table because characters are being stripped from the Shortcode.Inner.

As a minimal example this renders perfectly with a centred middle column.

```
| Column should align left | Column should align center | Column should align right |
|:----------- |:-------------:| ------------:|
| align: left | align: center | align: right |
```

And a shortcode that used `{{ .Inner | markdownify }}` would work fine in 0.54, but in 0.55+ the `.Inner` contents now elide colons separated by dashes, leaving the middle column missing any alignment or padding characters.

```
| Column should align left | Column should align center | Column should align right |
|:----------- || ------------:|
| align: left | align: center | align: right |
```

This results in invalid Markdown so instead of an HTML table the now-broken Markdown table is rendered instead.

BTW: The ```{{ $_hugo_config := `{ "version": 1 }`}}``` fix does not have any effect on this; rendering of the table is broken either way.

**Comment by yawpitch** [comment]
This is mentioned at the end of a Support topic [here](https://discourse.gohugo.io/t/hugo-0-55-breaks-shortcode/18315).

**Comment by bep** [comment]
Keep this discussion in one place.

**Comment by yawpitch** [comment]
I opened the issue here specifically because the discussion in the Support forum is far more prone to unhelpful tangents than here, and also because I have now identified one specific defect in Hugo 0.55+ that breaks shortcodes. I am certain it is not the only such defect, so it seemed sensible to open a specific issue for the first known defect for which a minimal reproducible case could be described.

The Contribution Guidelines may need some updating to reflect your preference for discussion to begin _and remain_ on the forums: I started with a general issue in the support forum, I then came here once I was "sure if it's a bug or not" for a specific sub-issue of that general one. I'd personally rather receive a notification when a specific Issue is addressed in a PR rather than wait on the Support forum, which seems to have an _extensive_ number of threads regarding bugs that have been addressed in later releases but which have never been updated to reflect that fact, but I can appreciate that some might prefer the forum format.
